### PR TITLE
Moves the plant analyzer chem mode to right click + lets you read out a tray's chemical contents

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Hot off the combat mode train, this PR moves the chemical scan of the plant analyzer from a toggle to right click.

This also moves the pod person chemical scan for plant analyzers to right click to match health analyzers.

I also moved all the various plant analyzer scans in the attackbys into the analyzer itself - I'm not sure if this is the better way to go about it or of there's better ways. (TOOL_PLANT_ANALYZER or something?)

Additionally, when scanning trays on chemical mode now, it'll read out the tray's chemical contents.
image

## Why It's Good For The Game

Combat mode is the sick new thing and this feels much better to use.

## Changelog
:cl: Melbert
add: Plant analyzers can now read out the reagent contents of a tray.
tweak: The chemical scan mode of plant analyzers was moved to right click, and is no longer a toggle.
/:cl:

### CODE REVIEW

File: code/_onclick/item_attack.dm [hunk 1]
```diff
@@ -38,7 +38,7 @@
 				attackby_result = target.attackby(src, user, params)
 			if (SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN)
 				return TRUE
-			if (SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN)
+			if (SECONDARY_ATTACK_CONTINUE_CHAIN)
```
Comment (MrMelbert) [inline code/_onclick/item_attack.dm:41]: This was requested by @Jared-Fogle because they did an oopsie.

File: code/modules/hydroponics/grown.dm [hunk 2]
```diff
@@ -80,14 +80,9 @@
...
-		var/obj/item/plant_analyzer/plant_analyzer = O
-		to_chat(user, plant_analyzer.scan_plant(src))
-	else
-		if(seed)
-			for(var/datum/plant_gene/trait/T in seed.genes)
-				T.on_attackby(src, O, user)
-
+	if(seed)
+		for(var/datum/plant_gene/trait/T in seed.genes)
+			T.on_attackby(src, O, user)
```
Comment (Jared-Fogle) [inline code/modules/hydroponics/grown.dm:85]:
```suggestion
		for(var/datum/plant_gene/trait/trait as anything in seed.genes)
			trait.on_attackby(src, O, user)
```

File: code/modules/hydroponics/hydroitemdefines.dm [hunk 3]
```diff
@@ -12,29 +12,87 @@
 	w_class = WEIGHT_CLASS_TINY
 	slot_flags = ITEM_SLOT_BELT
 	custom_materials = list(/datum/material/iron=30, /datum/material/glass=20)
+	/// The current scan mode of the plant analyzer. STATS on left click, CHEMS on right click.
```
Comment (Jared-Fogle) [inline code/modules/hydroponics/hydroitemdefines.dm:15]: Remove this variable entirely, and just call different procs.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Get rid of the mode variable.

File: code/modules/hydroponics/grown.dm (hunk 2 above)
Comment (MrMelbert) [reply code/modules/hydroponics/grown.dm:85]: This is getting removed entirely in this refactor #56841 , should I still make this change?

Comment (Jared-Fogle) [reply code/modules/hydroponics/grown.dm:85]: ehhhhhhhh leave this as a draft until that ones merged then

File: code/modules/hydroponics/hydroitemdefines.dm [hunk 4]
```diff
@@ -140,13 +142,13 @@
...
  */
-/obj/item/plant_analyzer/proc/scan_plant(obj/item/scanned_object)
+/obj/item/plant_analyzer/proc/scan_plant(obj/item/scanned_object, mode)
 	var/returned_message = "<span class='info'>*---------*\nThis is \a <span class='name'>[scanned_object]</span>.\n"
 	var/obj/item/seeds/our_seed = scanned_object
 	if(!istype(our_seed)) //if we weren't passed a seed, we were passed a plant with a seed
 		our_seed = scanned_object.get_plant_seed()
 
-	switch(scan_mode)
+	switch(mode)
```
Comment (Jared-Fogle) [inline code/modules/hydroponics/hydroitemdefines.dm:151]: I'd much rather these be separate procs. There's no reason to share this behavior in this one area.

File: code/modules/hydroponics/hydroitemdefines.dm [hunk 5]
```diff
@@ -12,86 +18,229 @@
...
+ * returns FALSE if it's something we can't scan.
+ * returns do_scan if it's something we CAN scan.
+ */
+/obj/item/plant_analyzer/proc/try_scan(atom/scan_target, mob/user, mode)
+	var/mob/living/living_user = user
+	if(living_user.combat_mode)
+		return FALSE
+
+	switch(mode)
+		if(PLANT_SCANMODE_STATS)
```
Comment (Jared-Fogle) [inline code/modules/hydroponics/hydroitemdefines.dm:50]: Get rid of these entirely, call the procs directly.

**Pull request merged**